frontend/src/components/pages/topics/TopicConfiguration.tsx

```diff
@@ -361,89 +360,73 @@ function isTopicConfigEdittable(
 }
 
-export const ConfigEntryEditor = observer(
-  (p: {
-    entry: ConfigEntryExtended;
-    className?: string;
-  }) => {
-    const entry = p.entry;
-    switch (entry.frontendFormat) {
-      case 'BOOLEAN':
-        return (
-          <SingleSelect
-            options={[
-              { value: 'false', label: 'False' },
-              { value: 'true', label: 'True' },
-            ]}
-            value={entry.currentValue}
-            onChange={(c) => (entry.currentValue = c)}
-          />
-        );
-
-      case 'SELECT':
-        return (
-          <SingleSelect
-            value={entry.currentValue}
-            onChange={(e) => (entry.currentValue = e)}
-            className={p.className}
-            options={
-              entry.enumValues?.map((value) => ({
-                value,
-                label: value,
-              })) ?? []
-            }
-          />
-        );
-
-      case 'BYTE_SIZE':
-        return (
-          <DataSizeSelect
-            allowInfinite={true}
-            valueBytes={Number(entry.currentValue ?? 0)}
-            onChange={(e) => (entry.currentValue = Math.round(e))}
-            className={p.className}
-          />
-        );
-      case 'DURATION':
-        return (
-          <DurationSelect
-            allowInfinite={true}
-            valueMilliseconds={Number(entry.currentValue ?? 0)}
-            onChange={(e) => (entry.currentValue = Math.round(e))}
-            className={p.className}
-          />
-        );
-
-      case 'PASSWORD':
-        return (
-          <PasswordInput value={entry.currentValue ?? ''} onChange={(x) => (entry.currentValue = x.target.value)} />
-        );
-
-      case 'RATIO':
-        return <RatioInput value={Number(entry.currentValue)} onChange={(x) => (entry.currentValue = x)} />;
-
-      case 'INTEGER':
-        return (
-          <NumInput
-            value={Number(entry.currentValue)}
-            onChange={(e) => (entry.currentValue = Math.round(e ?? 0))}
-            className={p.className}
-          />
-        );
-
-      case 'DECIMAL':
-        return (
-          <NumInput
-            value={Number(entry.currentValue)}
-            onChange={(e) => (entry.currentValue = e)}
-            className={p.className}
-          />
-        );
-      default:
-        return <Input value={String(entry.currentValue)} onChange={(e) => (entry.currentValue = e.target.value)} />;
-    }
-  },
-);
+export const ConfigEntryEditorController = <T extends string | number>(p: {
+  entry: ConfigEntryExtended;
+  value: T;
+  onChange: (e: T) => void;
+  className?: string;
+}) => {
+  const { entry, value, onChange } = p;
+  switch (entry.frontendFormat) {
+    case 'BOOLEAN':
+      return (
+        <SingleSelect<T>
+          options={[
+            { value: 'false' as T, label: 'False' },
+            { value: 'true' as T, label: 'True' },
+          ]}
+          value={value}
+          onChange={onChange}
+        />
+      );
```

Comment:
Any reason why we are not just using a Checkbox/Toggle/Switch components here?

Reply:
I realise it's code from before but just curious if you may know why

Reply:
I don't know, it's a bit out of scope of this PR, but I'm happy to refactor this if we want to have this cleaner.